examples/publisher_example.rs

```diff
@@ -0,0 +1,61 @@
+/********************************************************************************
```

Comment:
I like having these examples!

I think it'd be great to take these along with ones you can create for RPC and Notification and turn them into integration tests in the tests/ folder.